FreeRTOS/FreeRTOS
# Update cellular comm interface to generate interrupt from windows thread

Description
-----------
The API `vPortGenerateSimulatedInterruptFromWindowsThread()` added in https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/1044 enables the application to define it's own interrupt number and generated a simulated interrupt from windows thread. This reduce the synchronization effort between windows thread and FreeRTOS tasks.
In this PR:
* Use vPortGenerateSimulatedInterruptFromWindowsThread in the implementation. Creating FreeRTOS task is no longer required in the implementation.
* Rename prvSetupCommSettings to prvSetupCommState since SetCommState in called in this function.

Test Steps
-----------
Running one of the cellular interface demo for hours without problem.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
